orangecontrib/text/widgets/owdocumentembedding.py

```diff
@@ -0,0 +1,236 @@
+from AnyQt.QtWidgets import QApplication, QLayout, QPushButton, QStyle, QLabel
+from AnyQt.QtCore import Qt
+
+import numpy as np 
+
+from orangecontrib.text.widgets.utils import owbasevectorizer, widgets
+from Orange.widgets.gui import widgetBox, comboBox, spin, auto_commit, hBox
+from Orange.widgets.settings import Setting
+from Orange.widgets.widget import OWWidget, Msg, Input, Output
+from Orange.widgets.utils.concurrent import ConcurrentWidgetMixin, TaskState
+
+
+from orangecontrib.text.corpus import Corpus
+
+from orangecontrib.text.vectorization.dense import PretrainedEmbedder
+from orangecontrib.text.vectorization.dense import LANGS_TO_ISO, AGGREGATORS, EMB_DIM
+from Orange.misc.utils.embedder_utils import EmbeddingConnectionError
+from typing import Any
+
+def run_pretrained_embedder(
+                        corpus: Corpus,
+                        language: str, 
+                        aggregator: str, 
+                        state: TaskState) -> Corpus:
+
+    """Runs PretrainedEmbedder.
+
+    Args:
+        corpus (Corpus): Corpus on which transform is performed.
+        language (str): ISO 639-1 (two-letter) code of desired language.
+        aggregator (str): Aggregator which creates document embedding (single 
+                          vector) from word embeddings (multiple vectors). 
+                          Allowed values are mean, sum, max, min.
+        state (TaskState): State object.
+
+    Returns:
+        New corpus with additional features.
+    """
+
+    embedder = PretrainedEmbedder(language=language, 
+                                  aggregator=aggregator)
+    
+    ticks = iter(np.linspace(0., 100., len(corpus)))
+
+    def advance(success=True):
+        if(state.is_interruption_requested()):
+            embedder.set_cancelled()
+        if(success):
+            state.set_progress_value(next(ticks))
+
+    try:
+        new_corpus = embedder.transform(corpus, processed_callback=advance)
+        return new_corpus
+
+    except EmbeddingConnectionError:
+        return None
+
+
+class OWPretrainedEmbedder(OWWidget, ConcurrentWidgetMixin):
```

Comment:
This name should match to the widget name if widget name is Document Embedding then it should be OWDocumentEmbedding